app/src/main/java/com/github/mik629/android/fundamentals/ui/ViewState.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.mik629.android.fundamentals.ui
+
+
+/**
+ * Describes state of the view at any
+ * point of time.
+ */
+sealed class ViewState<out S : Any?, out E : Any?> {
+
+    data class Success<out S : Any?>(val result: S) : ViewState<S, Nothing>()
+
+    object Loading : ViewState<Nothing, Nothing>()
+
+    data class Error<out E : Any?>(val result: E) : ViewState<Nothing, E>()
+
+    companion object {
```

Comment:
why u need all this functions?
they just duplicate construnctor's calls
and u still can call construnctor

Reply:
sorry, copy-paste, thanks!